src/fixate/drivers/dmm/helper.py

```diff
@@ -101,3 +104,30 @@ def reset(self):
 
     def get_identity(self):
         raise NotImplementedError
+
+    @dataclass
+    class MeasurementStats:
+        min: float
+        max: float
+        avg: float
+
+    # context manager for setting NPLC
+    class _nplc_context_manager(object):
+        def __init__(self, dmm, nplc=None):
+            self.dmm = dmm
+            self.nplc = nplc
+            self.original_nplc = dmm.get_nplc()
+
+        def __enter__(self):
+            self.dmm.set_nplc(self.nplc)
+
+        # return to default NPLC setting
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            # check if an exception was raised
+            if exc_type is not None or exc_val is not None or exc_tb is not None:
```

Comment:
Pretty sure this guy can be reduced to:
```
if exc_type:
```

Reply:
Done as of 1fdc57f

Reply:
Shouldn't it be removed completely?

I think we want to restore the NPLC setting if there was an error and in normal operation

Reply:
We have to return false otherwise the context manager "swallows" the exception and fixate will not stop executing the test script.

however we could move the `self.dmm.set_nplc(self.original_nplc)` above this check. However, what happens if the reason an exception was thrown was because of an instrument error with the DMM?

Reply:
> We have to return false otherwise the context manager "swallows" the exception and fixate will not stop executing the test script.

I don't think that is quite correct. The docs say:

> If an exception is supplied, and the method wishes to suppress the exception (i.e., prevent it from being propagated), it should return a true value.

If we want to exception to be re-raised, we simply can do nothing and the implicit return value of `None` will result in the exception being propagated.

> However, what happens if the reason an exception was thrown was because of an instrument error with the DMM?

In this case we will just get a chained exception. https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

Reply:
I've removed the error checking as of 43b8675